internal/server/distribution.go

```diff
@@ -38,16 +38,20 @@ func ReadDistributions(distsDir, distro string) ([]DistributionFile, error) {
 			return nil
 		}
 
-		f, err := os.Open(path) // #nosec G304
+		cleanPath := filepath.Clean(path)
+		f, err := os.Open(cleanPath)
 		if err != nil {
 			return err
 		}
-		defer f.Close() // #nosec G307
 		var d DistributionFile
 		err = json.NewDecoder(f).Decode(&d)
 		if err != nil {
 			return err
 		}
+		err = f.Close()
```

Comment:
Hmm, if the decoding errors out the file will never be closed now. I think what's best is to change the defer function a little, see https://github.com/securego/gosec/issues/512#issuecomment-675286833 ; wdyt?

Reply:
Yes, that makes sense - thanks for pointing that out! I'll make changes